File: algorithms/0204.count_primes/count_primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        vector<bool> primes(n, true);
        int ans = 0;
        for (int64_t i = 2; i < n; ++i) {
            ans += static_cast<int>(primes[i]);
            if (i * i < n) {
                for (int64_t j = i * i; j < n; j += i) {
                    primes[j] = false;
                }
            }
        }
        return ans;
    }
};

File: algorithms/0204.count_primes/count_primes.rs
impl Solution {
    pub fn count_primes(n: i32) -> i32 {
        let n = n as usize;
        let mut is_primes = vec![false; n];
        let mut count = 0;
        for i in 2..n {
            if !is_primes[i] {
                count += 1;
                let mut j = (i << 1);
                while j < n {
                    is_primes[j] = true;
                    j += i;
                }
            }
        }
        count
    }
}